dipy/workflows/multi_io.py

```diff
@@ -0,0 +1,125 @@
+
+    if len(inputs) > 1:
+
+        if np.sum(np.diff(sizes_of_inputs)) == 0:
+            mixing_names = concatenate_inputs(inputs)
+
+            for (mix_inp, inp) in zip(mixing_names, inputs[0]):
+                dname = path.join(out_dir, path.dirname(inp))
+                updated_out_files = []
+                for out_file in out_files:
+                    updated_out_files.append(
+                        path.join(dname, mix_inp + '_' + out_file))
+                outputs.append(updated_out_files)
+
+        else:
+            max_size = np.max(sizes_of_inputs)
+            min_size = np.min(sizes_of_inputs)
+            if min_size > 1 and min_size != max_size:
+                raise ImportError('Size of input issue')
+            elif min_size == 1:
+                for i, sz in enumerate(sizes_of_inputs):
+                    if sz == min_size:
+                        inputs[i] = max_size * inputs[i]
+
+            mixing_names = concatenate_inputs(inputs)
+
+            for (mix_inp, inp) in zip(mixing_names, inputs[0]):
+                dname = path.join(out_dir, path.dirname(inp))
+                updated_out_files = []
+                for out_file in out_files:
+                    updated_out_files.append(
+                        path.join(dname, mix_inp + '_' + out_file))
+                outputs.append(updated_out_files)
+
+    elif len(inputs) == 1:
+
+        for inp in inputs[0]:
+            dname = path.join(out_dir, path.dirname(inp))
+            # base = path.splitext(path.basename(inp))[0]
+            base = basename(inp)
+            new_out_files = []
+            for out_file in out_files:
+                new_out_files.append(
+                    path.join(dname, base + '_' + out_file))
+            outputs.append(new_out_files)
+
+    return inputs, outputs
+
+
+def concatenate_inputs(multi_inputs):
+    """ Concatenate list of inputs
+    """
+
+    mixing_names = []
+    for inps in zip(*multi_inputs):
+        mixing_name = ''
+        for i, inp in enumerate(inps):
+            # mixing_name += path.splitext(path.basename(inp))[0]
+            mixing_name += basename(inp)
+            if i < len(inps) - 1:
+                mixing_name += '_'
+        mixing_names.append(mixing_name)
+    return mixing_names
+
+
+def basename(fname):
+    ext = path.splitext(path.basename(fname))[1]
+    base = path.splitext(path.basename(fname))[0]
+    if ext == '.gz':
+        ext = path.splitext(path.basename(base))[1]
+        if ext == '.nii':
+            base = path.splitext(path.basename(fname))[0]
+    return base
+
+
+class OutputCreator(object):
+    """ Create output filenames that work nicely with muiltiple input files from multiple directories (processing multiple subjects with one command)
+
+    Use information from input files, out_dir and out_fnames to generate correct outputs which can come from long lists of multiple or single
+    inputs.
+    """
+
+    def __init__(self):
```

Comment:
Since an OutputCreator always has to have its input and output set to work properly, should we add these as mandatory param to the __init__ ? This would make the OutputCreator object creation one line instead of the mandatory 3. 

Continuing in this line of thought, the next mandatory line for this object to be useful is create_outputs, IF we add inputs and outputs to init, we should also put create_ouputs in init since the object is just in a unusable state until we call this.